ArkGameExample/games/TankRaceGame/TankRaceEventHandler.swift

```diff
@@ -169,12 +169,10 @@ class TankRaceEventHandler {
         let eventData = event.eventData
         let tankEntity = eventData.tankEntity
         guard let tankHpComponent = context.ecs.getComponent(ofType: TankHpComponent.self, for: tankEntity),
-              tankHpComponent.hp <= 0,
-              var physicsComponent = context.ecs.getComponent(ofType: PhysicsComponent.self, for: tankEntity) else {
+              tankHpComponent.hp <= 0 else {
             return
         }
-        physicsComponent.toBeRemoved = true
-        context.ecs.upsertComponent(physicsComponent, to: tankEntity)
+        context.ecs.upsertComponent(ToRemoveComponent(toBeRemoved: true), to: tankEntity)
```

Comment:
we can perhaps make it more declarative by having a `Component.markForRemoval()` function, then within that function we upsert the `ToBeRemovedComponent` to the entity, then removing it